Automattic/jetpack
# Move logic to getMapProvider

Fixes https://github.com/Automattic/wp-calypso/issues/74048

## Proposed changes:
This PR moves the logic (show mapkit on WPCOM unless style is terrain or black&white) to the `getMapProvider() `function. The result of this function is now also passed down the component tree.

It also introduces a `get_map_provider()` function for the PHP-side, which does the same.

Lastly, it prevents the call to fetch the Mapbox API key when mapProvider is mapkit.

### Other information:


## Jetpack product discussion
pdDOJh-1Cb-p2#comment-1301

## Does this pull request change what data or activity we track or use?
No

## Testing instructions:
1. Apply this PR
2. Add a new post/page
3. Set the override cookie in your browser's console: `document.cookie = "map_provider=mapkit; path=/;";`
4. Add two map blocks: one default & one with the Terrain styling
5. The second one should load with Mapbox
6. Add a marker to the Mapkit one. The network tab should show a request that goes out to Apple.
7. Add a marker to the Mapbox one. The network tab should show a request that goes out to Apple.
8. Publish this post/page & view on the frontend
9. The bahvior should be the same
10. Now set the cookie to Mapbox: `document.cookie = "map_provider=mapbox; path=/;";`
11. Both maps should now use Mapbox on the front- & backend.
